velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -260,6 +264,64 @@ TEST_F(DateTimeFunctionsTest, dateAdd) {
   EXPECT_EQ(parseDate("1969-12-31"), dateAdd(parseDate("5881580-07-11"), kMin));
 }
 
+TEST_F(DateTimeFunctionsTest, dateAddSmallint) {
+  const auto dateAdd = [&](std::optional<int32_t> date,
+                           std::optional<int32_t> value) {
```

Comment:
Can we use int16_t type for value? Then `cast(c1 as smallint)` is not needed.

Reply:
I removed the cast in expression, but as the input data types are not same and we need to pass `DATE` type. so we need to construct row vector in lambda.
Thanks.